flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -1368,6 +1369,15 @@ private CompletedCheckpoint finalizeCheckpoint(PendingCheckpoint pendingCheckpoi
                                 CheckpointFailureReason.FINALIZE_CHECKPOINT_FAILURE, e1));
             }
 
+            if (e1 instanceof FlinkExpectedException) {
```

Comment:
I'm afraid `FlinkExcpectedException` will never reach this line
because it will be caught in `PendingCheckpoint.finalizeCheckpoint` and wrapped into `IOException` there.
Replacing `FlinkExpectedException` with `FlinkRuntimeException` shoudll be enough though.

This also suggests that some test is necessary.

Reply:
Updated the PR, For some reasons, I can't compile successfully in my local(debugging).